financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/common/ConsentExtensionUtils.java

```diff
@@ -137,4 +139,38 @@ public static <T> T  getClassInstanceFromFQN(String classpath, Class<T> classNam
             throw new ConsentManagementRuntimeException("Defined class" + classpath + "cannot be instantiated.", e);
         }
     }
+
+    /**
+     * Method to resolve username from user ID.
+     *
+     * @param userID   User ID
+     * @return Username
+     */
+    @Generated(message = "Ignoring because OAuth2Util cannot be mocked with no constructors")
+    public static String resolveUsernameFromUserId(String userID) {
+
+        if (!startsWithUUID(userID)) {
+            // If the user ID is not starting with a UUID that means request has sent the username,
+            // return the same user ID as the username.
+            return userID;
+        }
+
+        String username = null;
+        try {
+            if (userID.contains(ConsentExtensionConstants.TENANT_DOMAIN)) {
+                username =  OAuth2Util.resolveUsernameFromUserId(ConsentExtensionConstants.TENANT_DOMAIN,
+                        userID.split("@" + ConsentExtensionConstants.TENANT_DOMAIN)[0]);
+            } else {
+                username =  OAuth2Util.resolveUsernameFromUserId(ConsentExtensionConstants.TENANT_DOMAIN, userID);
+            }
+        } catch (UserStoreException e) {
+            return null;
```

Comment:
Shall we add a debug log here?

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/130/commits/1e13ec5af065388311c13d8ca7f04ef804d84184